docs/architecture/maui/mvvm.md

```diff
@@ -0,0 +1,364 @@
+
+With view model first composition, the app is conceptually composed of view models, with a service responsible for locating the view for a view model. View model first composition feels more natural to some developers, since the view creation can be abstracted away, allowing them to focus on the logical non-UI structure of the app. In addition, it allows view models to be created by other view models. However, this approach is often complex, and it can become difficult to understand how the various parts of the app are created and associated.
+
+> [!TIP]
+> Keep view models and views independent
+
+The binding of views to a property in a data source should be the view's principal dependency on its corresponding view model. Specifically, don't reference view types, such as Button and ListView, from view models. By following the principles outlined here, view models can be tested in isolation, therefore reducing the likelihood of software defects by limiting scope.
+
+The following sections discuss the main approaches to connecting view models to views.
+
+## Creating a view model declaratively
+
+The simplest approach is for the view to declaratively instantiate its corresponding view model in XAML. When the view is constructed, the corresponding view model object will also be constructed. This approach is demonstrated in the following code example:
+
+```xaml
+<ContentPage ... xmlns:local="clr-namespace:eShop">
+    <ContentPage.BindingContext>
+        <local:LoginViewModel />
+    </ContentPage.BindingContext>
+    ...
+</ContentPage>
+```
+
+When the ContentPage is created, an instance of the LoginViewModel is automatically constructed and set as the view's BindingContext.
+
+This declarative construction and assignment of the view model by the view has the advantage that it's simple, but has the disadvantage that it requires a default (parameter-less) constructor in the view model.
+
+## Creating a view model programmatically
+
+A view can have code in the code-behind file, resulting in the view model being assigned to its BindingContext property. This is often accomplished in the view's constructor, as shown in the following code example:
+
+```csharp
+public LoginView()
+{
+    InitializeComponent();
+    BindingContext = new LoginViewModel(navigationService);
+}
+```
+
+The programmatic construction and assignment of the view model within the view's code-behind has the advantage that it's simple. However, the main disadvantage of this approach is that the view needs to provide the view model with any required dependencies. Using a dependency injection container can help to maintain loose coupling between the view and view model. For more information, see [Dependency injection](dependency-injection.md).
```

Comment:
```suggestion
The programmatic construction and assignment of the view model within the view's code-behind has the advantage that it's simple. However, the main disadvantage of this approach is that the view needs to provide the view model with any required dependencies. Using a dependency injection container can help to maintain loose coupling between the view and view model. For more information, see [Dependency injection in .NET](../../core/extensions/dependency-injection.md).
```

Reply:
For this one, I will have a chapter in the book around dependency injection. I haven't kicked up a PR for it yet, so for this one, I believe, is a false positive.